### PR TITLE
gh-122571: Remove duplicated definitions of `PY_BUILTIN_HASHLIB_HASHES` in `configure.ac`

### PR DESCRIPTION
**Pull request by msprotz**
* Issue: gh-122571

### CODE REVIEW

File: Misc/NEWS.d/next/Build/2024-08-01-10-51-02.gh-issue-122571.fpA1xD.rst
Comment (picnixz) [inline Misc/NEWS.d/next/Build/2024-08-01-10-51-02.gh-issue-122571.fpA1xD.rst]: I'm not really sure whether a NEWS is needed, but if it is, it should be clearer (it's as if I was reading [X => Y => Z] + bad test).

Comment (msprotz) [reply Misc/NEWS.d/next/Build/2024-08-01-10-51-02.gh-issue-122571.fpA1xD.rst]: Personally I'd be in favor of no news, but I wanted to get CI green. If you can mark this PR as no NEWS entry needed, I'll remove the entry from the PR

Comment (picnixz) [reply Misc/NEWS.d/next/Build/2024-08-01-10-51-02.gh-issue-122571.fpA1xD.rst]: I'll do it (because the NEWS is more confusing that helping IMO...)

Comment (msprotz) [reply Misc/NEWS.d/next/Build/2024-08-01-10-51-02.gh-issue-122571.fpA1xD.rst]: Great, I force-pushed to remove the latest commit (NEWS entry). This PR is now a single commit that fixes configure.ac. Thank you.

File: configure.ac
Comment (erlend-aasland) [inline configure.ac]: This will not work; the variable `with_builtin_hashlib_hashes` is not defined yet.

**Review by erlend-aasland — CHANGES_REQUESTED** [review]

File: configure.ac
Comment (msprotz) [reply configure.ac]: I'm not really sure how to fix it then -- all I know is that the result of the previous style is two #defines in the conftest.h file... are you familiar with autoconf?

Comment (msprotz) [reply configure.ac]: (Oh and this change *did* fix my particular issue and seems to give the expected result, but I appreciate that this may not be the right way to go about it.)

Comment (erlend-aasland) [reply configure.ac]: When changing Autoconf input, also check the generated output. You're now writing `$with_builtin_hashlib_hashes` to `confdefs.h` before `with_builtin_hashlib_hashes` is set (it's set in the `AC_ARG_WITH` a few lines below).

The correct solution is to keep only the latter `AC_DEFINE_UNQUOTED` in L7423 and remove the one in L7406.

> Oh and this change did fix my particular issue and seems to give the expected result,

I suspect you're not getting the expected result. The expected result is a CSV C string literal with a list of the buildin hashes, but the actual result of this PR is that the macro is defined to the empty string:

```sh
$ ./configure
$ grep PY_BUILTIN_HASHLIB_HASHES pyconfig.h                  
#define PY_BUILTIN_HASHLIB_HASHES ""
```

My expected result would be:

```sh
$ ./configure
$ grep PY_BUILTIN_HASHLIB_HASHES pyconfig.h
#define PY_BUILTIN_HASHLIB_HASHES "md5,sha1,sha2,sha3,blake2"
```

Comment (msprotz) [reply configure.ac]: Yeah you are correct, this was definitely not working as intended. Thanks, this is now fixed.

**Comment by msprotz** [comment]
I have made the requested changes; please review again

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks, this looks good to me! I won't be able to verify it on my laptop until tomorrow evening, so I'll hold off merging until then.

**Comment by msprotz** [comment]
> Thanks, this looks good to me! I won't be able to verify it on my laptop until tomorrow evening, so I'll hold off merging until then.

No problem, thanks for the prompt review!

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thank you so much for finding and fixing this, @msprotz!

**Comment by erlend-aasland** [comment]
I'm not sure we should backport this to 3.12.

**Pull request merged**
